### PR TITLE
EWPP-768: Adds Description List field type.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: src/Plugin/Field/FieldType/DescriptionListFieldItem.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        'term' => [
+          'type' => 'text',
+          'size' => 'big',
+        ],
+        'description' => [
+          'type' => 'text',
+          'size' => 'big',
+        ],
+        'format' => [
+          'type' => 'varchar_ascii',
```
Comment (upchuk) [inline src/Plugin/Field/FieldType/DescriptionListFieldItem.php]: Why ascii? Didn't we run into trouble with this elsewhere? Let's use varchar

File: templates/description-list.html.twig [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * @file
+ * Description list template.
+ */
+#}
+
+{% if items %}
+  <dl>
+    {% for item in items %}
+      <dt>{{ item.term }}</dt>
+      <dd>{{ item.description }}</dd>
```
Comment (sergepavle) [inline templates/description-list.html.twig]: We may also not to output an empty dt/dd tags if the values are empty.
But it is an optional comment.

File: tests/src/Functional/DescriptionListFieldWidgetTest.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    $this->assertSession()->pageTextContains('Page My page has been created');
+
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $this->container->get('entity_type.manager')->getStorage('node')->load(1);
+    $expected_values = [
+      'term' => 'Term 1',
+      'description' => 'Description 1',
+      'format' => 'plain_text',
+    ];
+    $this->assertEquals($expected_values, $node->get('description_list')->first()->getValue());
```
Comment (sergepavle) [inline tests/src/Functional/DescriptionListFieldWidgetTest.php]: It would be good to add tests with 'Add another item' for testing multiple fields.

**Review by sergepavle — DISMISSED** [review]

**Review by sergepavle — DISMISSED** [review]

File: src/Plugin/Field/FieldType/DescriptionListFieldItem.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      ->setSetting('text source', 'description')
+      ->setInternal(FALSE);
+
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applyDefaultValue($notify = TRUE) {
```
Comment (upchuk) [inline src/Plugin/Field/FieldType/DescriptionListFieldItem.php]: Do we need this?

File: src/Plugin/Field/FieldWidget/DescriptionListFieldWidget.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    return $values;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function errorElement(array $element, ConstraintViolationInterface $violation, array $form, FormStateInterface $form_state) {
+    if (!empty($violation->arrayPropertyPath) && $sub_element = NestedArray::getValue($element, $violation->arrayPropertyPath)) {
+      return $sub_element;
```
Comment (upchuk) [inline src/Plugin/Field/FieldWidget/DescriptionListFieldWidget.php]: Do we have subelements here? Why is this needed? Also, if really needed, let's add a comment here describing a bit this rather cryptic logic.

File: templates/description-list.html.twig [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+ * @file
+ * Description list template.
+ */
+#}
+
+{% if items %}
+  <dl>
+    {% for item in items %}
+      {% if item.term %}
+        <dt>{{ item.term }}</dt>
```
Comment (upchuk) [inline templates/description-list.html.twig:12]: This doesn't really work. I left one term empty and it printed the <dt>:

Screen Shot 2021-02-24 at 14 26 20

It means it's missing also some test coverage.

**Review by upchuk — CHANGES_REQUESTED** [review]
Let's use percona 5.7 like everywhere else

**Review by upchuk — DISMISSED** [review]

File: templates/description-list.html.twig [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * @file
+ * Description list template.
+ */
+#}
+
+{% if items %}
+  <dl>
+    {% for item in items %}
+      {% if item.term['#plain_text'] is not empty %}
```
Comment (sergepavle) [inline templates/description-list.html.twig]:
```suggestion
      {% if item.term['#plain_text'] %}
```

File: templates/description-list.html.twig [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        <dt>{{ item.term }}</dt>
+      {% endif %}
+      {% if item.description['#text'] is not empty %}
```
Comment (sergepavle) [inline templates/description-list.html.twig]:
```suggestion
      {% if item.description['#text'] %}
```

**Review by upchuk — APPROVED** [review]

**Review by sergepavle — APPROVED** [review]

**Pull request merged**